### ISSUE TITLE
`ExchangeBuilder` for building an exchange transaction

### ISSUE THREAD

**Issue by homura** [ckb-js/ckit#108]
## Motivation

As a sUDT holder, I want to send a transaction like depositing my sUDT to Godwoken, but I don't have enough CKB to create the Godwoken deposit cell. If the ckit library provide an `ExchangeBuilder`, so that I can exchange my sUDT for CKB and make the deposition.

The diagram below is an example of exchanging my 1 unit of sUDT for 379 CKB to deposit to Godwoken

```mermaid
graph LR;
  ExchangeProviderO[ExchangeProvider']
  HolderSUDTO[HolderSUDT']

  subgraph inputs
  ExchangeProvider
  HolderSUDT
  end

  subgraph outputs
  ExchangeProviderO
  HolderSUDTO
  GodwokenDeposit
  end
  
  ExchangeProvider -.-> ExchangeProviderO;
  ExchangeProvider -- +379 CKB --> GodwokenDeposit;
  HolderSUDT -.-> HolderSUDTO;
  HolderSUDT -- sudtAmountForExchange 1 SUDT --> ExchangeProviderO;
  HolderSUDT -- sudtAmountForRecipient 1 SUDT --> GodwokenDeposit;
```

### Exchange Provider

```ts
interface ExchangeSudtForCkbOptions {
  // target sudt
  sudt: CkbTypeScript;
  sudtSender: Address;
  sudtAmountForExchange: HexNumber;
  sudtAmountForRecipient: HexNumber;

  // the lock(address) of exchange provider
  exchangeProvider: Cell[] | Address;
  ckbAmountForRecipient: HexNumber;

  // receive the ckbAmount + sudtAmountForRecipient
  exchangeRecipient: Address;
}

const builder = new ExchangeSudtForCkbBuilder(...);
const exchangeTx = await builder.build();

// signed by the exchange provider
const partialSignedTx = exchangeProviderSigner.seal(exchangeTx);

const serializedTx = serde.serialize(partialSignedTx)

// exchange provider send the serialized partial signed transaction to the sUDT holder...
```

### SUDT Holder

```ts
// the sUDT holder received the serialized partial signed transaction

const partialSignedTx = serde.deserialize(serializedTx);

// signed by the sUDT holder
const fullySignedTx = holderSigner.seal(partialSignedTx);

provider.sendTransaction(fullySignedTx);

```

**Comment by homura** [comment]
update: change type of `exchangeProvider` from `Address` to `Cell[] | Address` to make the exchange cells can be controlled by the user

**Comment by homura** [comment]
Resolved by #109
